packages/lib/src/vis/shared/SelectionMesh.tsx

```diff
@@ -0,0 +1,53 @@
+import { useThree } from '@react-three/fiber';
+import type { ThreeEvent } from '@react-three/fiber';
+import { useCallback, useState } from 'react';
+import { Vector2 } from 'three';
+
+import Selection from './Selection';
+
+function SelectionMesh() {
+  const { width, height } = useThree((state) => state.size);
+  const [startPoint, setStartPoint] = useState<Vector2>();
+  const [endPoint, setEndPoint] = useState<Vector2>();
+  const [isDragging, setDrag] = useState(false);
+
+  const onPointerDown = useCallback((evt: ThreeEvent<PointerEvent>) => {
+    const { target, pointerId, unprojectedPoint } = evt;
+    (target as Element).setPointerCapture(pointerId);
+    setStartPoint(new Vector2(unprojectedPoint.x, unprojectedPoint.y));
+    setEndPoint(new Vector2(unprojectedPoint.x, unprojectedPoint.y));
+    setDrag(true);
+  }, []);
+
+  const onPointerMove = useCallback(
+    (evt: ThreeEvent<PointerEvent>) => {
+      if (!isDragging) {
+        return;
+      }
+      setEndPoint(new Vector2(evt.unprojectedPoint.x, evt.unprojectedPoint.y));
+    },
+    [isDragging]
+  );
+
+  const onPointerUp = useCallback((evt: ThreeEvent<PointerEvent>) => {
+    const { target, pointerId, unprojectedPoint } = evt;
+
+    (target as Element).releasePointerCapture(pointerId);
+    setEndPoint(new Vector2(unprojectedPoint.x, unprojectedPoint.y));
+    setDrag(false);
+  }, []);
+
+  return (
+    <>
+      <mesh {...{ onPointerMove, onPointerUp, onPointerDown }}>
+        <meshBasicMaterial opacity={0} transparent color="#0000ff" />
+        <planeGeometry args={[width, height]} />
```

Comment:
We actually have a `VisMesh` for this :wink: -- which, I'm noticing, we should use in `PanZoomMesh` too.

Reply:
Good point !

I am not sure for `PanZoomMesh` though: `VisMesh` covers only the vis space and not the canvas space which would prevent from panning/zooming when the mouse is outside the vis but in the canvas.